maintenance/insertMissingDumps.php

```diff
@@ -34,39 +36,61 @@ public function execute() {
 		$dumpFiles = iterator_to_array( $backend->getFileList( [
 			'dir' => $storagePath,
 			'adviseStat' => true,
-			'topOnly' => true
+			'topOnly' => true,
 		] ) );
 
-		$missingDumps = array_diff( array_values( $dumpFiles ), $existingDumps );
-		foreach ( $missingDumps as $dump ) {
-			$fileSize = $backend->getFileSize( [
-				'src' => "$storagePath/$dump"
-			] );
+		$chunkedFiles = [];
+		foreach ( $dumpFiles as $file ) {
+			// Group files by their base names (strip .part<number>).
+			if ( preg_match( '/^(.*)\.part\d+$/', $file, $matches ) ) {
```

Comment:
```suggestion
			if ( preg_match( '/\.part\d+$/', $file ) ) {
```

Reply:
Oh my god, I missed that part somehow

Reply:
Then matches doesn't output what we need it to, it matches match groups so like those in parentheses first group is $matches[1]

Reply:
oh my god i am embarassing myself so much :neocat_blush_hide: